lttoolbox/compiler.cc

```diff
@@ -208,13 +212,20 @@ Compiler::procParDef()
   if(type != XML_READER_TYPE_END_ELEMENT)
   {
     current_paradigm = attrib(COMPILER_N_ATTR);
+	current_minimise = true;
+	if (attrib(COMPILER_REGEX_ATTR) == L"yes") {
+	  current_minimise = false;
+	}
   }
   else
   {
     if(!paradigms[current_paradigm].isEmpty())
     {
-      paradigms[current_paradigm].minimize();
-      paradigms[current_paradigm].joinFinals();
+	  if (current_minimise) {
+		paradigms[current_paradigm].minimize();
+	  } else {
+		paradigms[current_paradigm].joinFinals();
+	  }
```

Comment:
Shouldn't this be `if(current_minimise) minimize(); joinFinals()` , ie. we want to joinFinals also when we minimize? (Seems like the current change would also alter how non-paradigms are compiled!)

Reply:
`minimize()` calls `reverse()` which calls `joinFinals()`, so there shouldn't be a change in behavior.

Reply:
Calls `reverse` twice in fact, so it both joins finals and initials … I suppose nowhere in that sequence any new (unjoined) final is added so it's safe then :)